sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/json/JsonSuite.scala

```diff
@@ -3450,6 +3450,174 @@ abstract class JsonSuite
     }
   }
 
+  test("SPARK-44940: fully parse the record except f1 if partial results are enabled") {
```

Comment:
just to confirm: this PR is only for perf and we add more tests because the original PR that added partial result handling did not add enough tests?

Reply:
Not really, the PR also fixes a ClassCastException error, which should have been fixed when I added the original code for partial results parsing.